tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,737 @@
+                    .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                    .build();
+            FileSystemRealmUtil.createEncryptedRealmFromUnencrypted(oldFileSystemRealm, newFileSystemRealm);
+        }
+    }
+
+    /**
+     * Creates the script/commands the user must run for Elytron to recognize
+     * and use the new filesystem-realm
+     */
+
+    private void createWildFlyScript() throws Exception {
+        int counter = 0;
+        for (Descriptor descriptor : descriptors) {
+            String inputRealmLocation = descriptor.getInputRealmLocation();
+            String outputRealmLocation = descriptor.getOutputRealmLocation();
+            String fileSystemRealmName = descriptor.getFileSystemRealmName();
+            String credentialStore = descriptor.getCredentialStore();
+            String secretKeyAlias = descriptor.getSecretKeyAlias();
+            int levels = descriptor.getLevels();
+            Charset hashCharset = descriptor.getHashCharset();
+
+            if(secretKeyAlias == null) {
+                secretKeyAlias = "key";
+            }
+            String createScriptCheck = "";
+            if (Paths.get(String.format("%s.sh", fileSystemRealmName)).toFile().exists()) {
+                createScriptCheck = prompt(
+                        false,
+                        null,
+                        false,
+                        ElytronToolMessages.msg.shouldFileBeOverwritten(String.format("%s.sh", fileSystemRealmName))
+                );
+            }
+            String fullOutputPath;
+            if (outputRealmLocation.startsWith(".")) {
+                fullOutputPath = Paths.get(outputRealmLocation.substring(2, outputRealmLocation.length())).toAbsolutePath().toString();
+            } else {
+                fullOutputPath = Paths.get(outputRealmLocation).toAbsolutePath().toString();
+            }
+
+            if (summaryMode) {
+                summaryString.append(String.format("Configured script for WildFly named %s.sh at %s.", fileSystemRealmName, fullOutputPath));
+                summaryString.append(System.getProperty("line.separator"));
+                summaryString.append("The script is using the following names:");
+                summaryString.append(System.getProperty("line.separator"));
+                summaryString.append(String.format("Name of filesystem-realm: %s", fileSystemRealmName));
+                summaryString.append(System.getProperty("line.separator"));
+            }
+
+            List<String> scriptLines = Arrays.asList(
+                String.format("/subsystem=elytron/secret-key-credential-store=%s:add(path=%s, create=true)", "mycredstore"+counter, fullOutputPath),
+                String.format("/subsystem=elytron/filesystem-realm=%s:add(path=%s, levels=%s, hash-hashCharset=%s, credential-store=%s, secret-key=%s)", fileSystemRealmName, levels, fullOutputPath, hashCharset.toString(), "mycredstore"+counter, secretKeyAlias)
```

Comment:
@Ashpan just minor but the levels, fullOutputPat are switched here

Reply:
@Ashpan I have unencrypted filesystem-realm, and I manually run `filesystem-realm-encrypt` command with no credential store. Then I add my new encrypted realm to the subsystem with these (fixed) commands, and I try: `encrypted-filesystem-realm:read-identity(identity=sara)` it says `"WFLYELY01002: Identity with name [sara] not found.",` even though `sara` is in the unencrypted store and I can see `ONQXEYI.xml` file being in the new encrypted store. Not sure if I have some problem locally or these commands are missing something.

Reply:
@Ashpan Just noticed the fullOutputPath is missing the name of the filesystem store, it should be `fullOutputPath/fileSystemRealmName `

Reply:
Fixed